lib/oxrdf/src/literal.rs

```diff
@@ -159,6 +161,63 @@ impl fmt::Display for Literal {
     }
 }
 
+#[cfg(feature = "serde")]
+impl Serialize for Literal {
+    #[inline]
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut map: <S as Serializer>::SerializeStruct;
+        match self {
+            Self(LiteralContent::String(value)) => {
+                map = serializer.serialize_struct("Literal", 1)?;
+                map.serialize_field("value", value)?;
+            }
+            Self(LiteralContent::LanguageTaggedString { value, language }) => {
+                map = serializer.serialize_struct("Literal", 2)?;
+                map.serialize_field("value", value)?;
+                map.serialize_field("language", language)?;
+            }
+            Self(LiteralContent::TypedLiteral { value, datatype }) => {
+                map = serializer.serialize_struct("Literal", 2)?;
+                map.serialize_field("value", value)?;
+                map.serialize_field("datatype", datatype.as_str())?;
+            }
+        };
+        map.end()
+    }
+}
+
+#[cfg(feature = "serde")]
+#[derive(Deserialize)]
+struct LiteralValue {
+    value: String,
+    language: Option<String>,
+    datatype: Option<String>,
+}
```

Comment:
hyper-nit: you can move this struct inline in the `deserialize` function to make clear it's only a temporary helper